packages/mds-api-server/index.ts

```diff
@@ -172,14 +172,39 @@ export const ApiVersionMiddleware = <TVersion extends string>(mimeType: string,
       }))
       .filter(info => info.latest !== undefined)
 
-    // Get supported version with highest q value
-    if (supported.length > 0) {
+    if (req.method === 'OPTIONS') {
+      /* If the incoming request is an OPTIONS request,
+       * immediately respond with the negotiated version.
+       * If the client did not negotiate a valid version, fall-through to a 406 response.
+       */
+      if (supported.length > 0) {
+        const [{ latest }] = supported.sort((a, b) => b.q - a.q)
```

Comment:
Nit but it seems to me that you could factor this out of the two branches of this `if` statement.

Reply:
Yeah I was thinking about that too, there's not a super clean way to structure this logic IMO. On one hand, being concise is ideal (though you can't be very concise here), on the other hand, being expressive is ideal. I'm gonna think on this a bit before merging and see if I can find a really clean way of writing it.

Reply:
IMHO "expressive" >> "concise"